src/net_processing.cpp

```diff
@@ -584,76 +578,105 @@ bool ProcessMessage(CNode *pfrom, std::string strCommand, CDataStream &vRecv, in
                 pfrom->fDisconnect = true;
             }
         }
-        pfrom->state_incoming = ConnectionStateIncoming::SENT_VERACK_READY_FOR_POTENTIAL_XVERSION;
     }
 
-    /* Since we are processing messages in multiple threads, we may process them out of order.  Does enforcing this
-       order actually matter?  Note we allow mis-order (or no version message at all) if whitelisted...
-        else if (pfrom->nVersion == 0 && !pfrom->fWhitelisted)
+    else if (strCommand == NetMsgType::XVERSION)
+    {
+        if (pfrom->fSuccessfullyConnected == true)
         {
-            // Must have version message before anything else (Although we may send our VERSION before
-            // we receive theirs, it would not be possible to receive their VERACK before their VERSION).
+            dosMan.Misbehaving(pfrom, 1);
```

Comment:
nit: why add the misbehavior here when on the next line you disconnect the peer?